File: app/src/main/java/com/zhuzichu/orange/ApplicationOrange.java
package com.zhuzichu.orange;

import android.app.Application;
import android.content.res.Configuration;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatDelegate;

import com.zhuzichu.base.ext.ToastExtKt;
import com.zhuzichu.base.global.AppGlobal;

import io.reactivex.plugins.RxJavaPlugins;
import me.jessyan.autosize.AutoSize;
import me.jessyan.autosize.AutoSizeConfig;

/**
 * @author Administrator
 */
public class ApplicationOrange extends Application {



    @Override
    public void onCreate() {
        super.onCreate();
        AppGlobal.INSTANCE.init(this);
        initAutoSize();
        RxJavaPlugins.setErrorHandler(throwable -> {

        });
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);

    }

    private void initAutoSize() {
        AutoSizeConfig.getInstance()
                .setCustomFragment(false)
                .setLog(false)
                .setUseDeviceSize(false)
                .setBaseOnWidth(true);
        AutoSize.initCompatMultiProcess(this);
    }

}

File: app/src/main/java/com/zhuzichu/orange/ApplicationOrange.kt
package com.zhuzichu.orange

import android.os.StrictMode
import android.os.StrictMode.ThreadPolicy.Builder
import androidx.appcompat.app.AppCompatDelegate
import com.zhuzichu.base.common.prefs.UserStorage

import com.zhuzichu.base.global.AppGlobal
import com.zhuzichu.orange.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

import io.reactivex.plugins.RxJavaPlugins.setErrorHandler
import javax.inject.Inject
import timber.log.Timber

class ApplicationOrange : DaggerApplication() {

    @Inject
    lateinit var userStorage: UserStorage

    override fun onCreate() {
        if (BuildConfig.DEBUG) {
            enableStrictMode()
            Timber.plant(Timber.DebugTree())
        } else {
            Timber.plant(CrashReportingTree())
        }
        super.onCreate()
        AppGlobal.init(this)
        setErrorHandler {

        }
        AppCompatDelegate.setDefaultNightMode(userStorage.uiMode)
    }

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerAppComponent.factory().create(this)
    }

    private fun enableStrictMode() {
        StrictMode.setThreadPolicy(
            Builder()
                .detectDiskReads()
                .detectDiskWrites()
                .detectNetwork()
                .penaltyLog()
                .build()
        )
    }

    private class CrashReportingTree : Timber.Tree() {
        override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {

        }
    }
}